data_io.py

```diff
@@ -0,0 +1,70 @@
+import mimetypes
+import os
+from json import dump
+
+import pandas as pd
+
+
+# Input
+def take_input_data(path: str) -> pd.DataFrame:
+    if (
```

Comment:
it is called LBYL (Look Before You Leap), it would be better to use EAFP (Easier to Ask For Forgiveness Than Permission). Check what errors are raised when for example file is empty, has improper data inside or is corrupted and process these appropriately.

Reply:
Disagree. Project requirements clearly put emphasis on  if - else if - else logic. We operate on plain/text and folders, don't care about other. That kind of analysis you are talking about can be done inside pandas dataframe. For simplicity I'm omitting it with on_bad_lines="skip" parameter-argument. Proper data clearing would be a separate project basically, fortunately I don't see it in requirements.

Reply:
OK wyszło na Twoje, plus zawiłości jak u Regiego, widać jednak doświadczenie.